### DISCUSSION TITLE
malamute  error

### DISCUSSION THREAD

**Discussion by Salma-Mao** [idaholab/moose#26183] [Q&A Getting Started] [answered]
hello all,
     I have installed the moose by docker. And I pull the malamute by container of docker. But I get the error. 

root@8a641d28a674:~/projects/malamute# git checkout main
Branch 'main' set up to track remote branch 'main' from 'origin'.
Switched to a new branch 'main'
root@8a641d28a674:~/projects/malamute# git submodule update --init moose
Submodule 'moose' (https://github.com/idaholab/moose.git) registered for path 'moose'
Cloning into '/root/projects/malamute/moose'...
Submodule path 'moose': checked out '2ad125da4dc5a8d5413806b0f3f35d7b46c7e7e5'
root@8a641d28a674:~/projects/malamute# ls
LICENSE  Makefile  NOTICE.txt  README.md  doc  examples  include  moose  run_tests  scripts  src  test  testroot  unit
root@8a641d28a674:~/projects/malamute# make -j 2
Checking if header needs updating: /opt/moose/framework/include/base/MooseRevision.h...
Linking Library /opt/moose/modules/navier_stokes/lib/libnavier_stokes-opt.la...
Linking Library /opt/moose/modules/contact/lib/libcontact-opt.la...
Linking Library /opt/moose/modules/phase_field/lib/libphase_field-opt.la...
Rebuilding symlinks in /root/projects/malamute/build/header_symlinks
Creating Unity Directory /root/projects/malamute/build/unity_src
Creating Unity /root/projects/malamute/build/unity_src/bcs_Unity.C
Creating Unity /root/projects/malamute/build/unity_src/indicators_Unity.C
Creating Unity /root/projects/malamute/build/unity_src/materials_Unity.C
Creating Unity /root/projects/malamute/build/unity_src/interfacekernels_Unity.C
Creating Resource file /root/projects/malamute/malamute.yaml
Checking if header needs updating: /root/projects/malamute/include/base/MalamuteRevision.h...
Creating Unity /root/projects/malamute/build/unity_src/kernels_Unity.C
Compiling C++ (in opt mode) /root/projects/malamute/build/unity_src/bcs_Unity.C...
Compiling C++ (in opt mode) /root/projects/malamute/build/unity_src/indicators_Unity.C...
Compiling C++ (in opt mode) /root/projects/malamute/build/unity_src/materials_Unity.C...
Compiling C++ (in opt mode) /root/projects/malamute/build/unity_src/interfacekernels_Unity.C...
In file included from /root/projects/malamute/build/unity_src/materials_Unity.C:3:
/root/projects/malamute/src/materials/ADGaussianHeatSourceBase.C: In static member function 'static InputParameters ADGaussianHeatSourceBase::validParams()':
/root/projects/malamute/src/materials/ADGaussianHeatSourceBase.C:26:96: error: no matching function for call to 'InputParameters::addParam<std::vector<double, std::allocator<double> > >(const char [2], <brace-enclosed initializer list>, const char [152])'
   26 |                                      "the effective radius to be equal along three directions.");
      |                                                                                                ^
In file included from /opt/moose/framework/build/header_symlinks/MooseObject.h:13,
                 from /opt/moose/framework/build/header_symlinks/MaterialBase.h:13,
                 from /opt/moose/framework/build/header_symlinks/Material.h:13,
                 from /root/projects/malamute/build/header_symlinks/ADGaussianHeatSourceBase.h:12,
                 from /root/projects/malamute/build/header_symlinks/ADFunctionPathGaussianHeatSource.h:12,
                 from /root/projects/malamute/src/materials/ADFunctionPathGaussianHeatSource.C:10,
                 from /root/projects/malamute/build/unity_src/materials_Unity.C:2:
/opt/moose/framework/build/header_symlinks/InputParameters.h:1359:1: note: candidate: 'template<class T, class S> void InputParameters::addParam(const string&, const S&, const string&)'
 1359 | InputParameters::addParam(const std::string & name, const S & value, const std::string & doc_string)
      | ^~~~~~~~~~~~~~~
/opt/moose/framework/build/header_symlinks/InputParameters.h:1359:1: note:   template argument deduction/substitution failed:
In file included from /root/projects/malamute/build/unity_src/materials_Unity.C:3:
/root/projects/malamute/src/materials/ADGaussianHeatSourceBase.C:26:96: note:   couldn't deduce template parameter 'S'
   26 |                                      "the effective radius to be equal along three directions.");
      |                                                                                                ^
In file included from /opt/moose/framework/build/header_symlinks/MooseObject.h:13,
                 from /opt/moose/framework/build/header_symlinks/MaterialBase.h:13,
                 from /opt/moose/framework/build/header_symlinks/Material.h:13,
                 from /root/projects/malamute/build/header_symlinks/ADGaussianHeatSourceBase.h:12,
                 from /root/projects/malamute/build/header_symlinks/ADFunctionPathGaussianHeatSource.h:12,
                 from /root/projects/malamute/src/materials/ADFunctionPathGaussianHeatSource.C:10,
                 from /root/projects/malamute/build/unity_src/materials_Unity.C:2:
/opt/moose/framework/build/header_symlinks/InputParameters.h:1378:1: note: candidate: 'void InputParameters::addParam(const string&, const string&) [with T = std::vector<double>; std::string = std::__cxx11::basic_string<char>]'
 1378 | InputParameters::addParam(const std::string & name, const std::string & doc_string)
      | ^~~~~~~~~~~~~~~
/opt/moose/framework/build/header_symlinks/InputParameters.h:1378:1: note:   candidate expects 2 arguments, 3 provided
make: *** [/opt/moose/framework/build.mk:144: /root/projects/malamute/build/unity_src/materials_Unity.x86_64-pc-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....

**Comment by GiudGiud** [comment, marked as answer]
Hello

This looks like the moose version is older than the malamute version. 

Make sure to use the moose submodule in malamute, that one will work for sure

Guillaume

**Reply by Salma-Mao** [reply]
Hello
       Thank you for your help. The problem have been solved.
